internal/config/config.go

```diff
@@ -106,3 +114,92 @@ func authenticationOption(clientID, clientSecret, apiToken, audience string) man
 
 	return management.WithClientCredentials(ctx, clientID, clientSecret)
 }
+
+func customClientWithRetries() *http.Client {
+	client := &http.Client{
+		Transport: rateLimitTransport(
+			retryableErrorTransport(
+				http.DefaultTransport,
+			),
+		),
+	}
+
+	return client
+}
+
+func rateLimitTransport(tripper http.RoundTripper) http.RoundTripper {
+	return rehttp.NewTransport(tripper, rateLimitRetry, rateLimitDelay)
+}
+
+func rateLimitRetry(attempt rehttp.Attempt) bool {
+	if attempt.Response == nil {
+		return false
+	}
+
+	return attempt.Response.StatusCode == http.StatusTooManyRequests
+}
+
+func rateLimitDelay(attempt rehttp.Attempt) time.Duration {
+	resetAt := attempt.Response.Header.Get("X-RateLimit-Reset")
+
+	resetAtUnix, err := strconv.ParseInt(resetAt, 10, 64)
+	if err != nil {
+		resetAtUnix = time.Now().Add(5 * time.Second).Unix()
+	}
+
+	return time.Duration(resetAtUnix-time.Now().Unix()) * time.Second
+}
+
+func retryableErrorTransport(tripper http.RoundTripper) http.RoundTripper {
+	return rehttp.NewTransport(
+		tripper,
+		rehttp.RetryAll(
+			rehttp.RetryMaxRetries(3),
+			rehttp.RetryAny(
+				rehttp.RetryStatuses(
+					http.StatusServiceUnavailable,
+					http.StatusInternalServerError,
+					http.StatusBadGateway,
+					http.StatusGatewayTimeout,
+					// Cloudflare-specific server error that is generated
+					// because Cloudflare did not receive an HTTP response
+					// from the origin server after an HTTP Connection was made.
+					524,
```

Comment:
This happened in here: https://github.com/auth0/terraform-provider-auth0/actions/runs/5974520479/job/16208826703#step:4:302.